### PR TITLE
feat: Allow reading from a custom cfg file path

### PR DESCRIPTION
**Pull request by jasonz-dfinity**
Allow a custom cfg file path and a custom drun path in the cfg file. This is to make it possible for bazel to run canbench.

### CODE REVIEW

File: canbench-bin/src/main.rs [hunk 1]
```diff
@@ -19,20 +19,28 @@ struct Args {
     // If true, only prints the benchmark results (and nothing else).
     #[clap(long)]
     less_verbose: bool,
+
+    // If provided, use the specified configuration file instead of the default one.
+    #[clap(long, value_parser = value_parser!(PathBuf))]
+    cfg_file_path: Option<PathBuf>,
```
Comment (ielashi) [inline canbench-bin/src/main.rs:25]: nit: for brevity, I'd shorten the name.

```suggestion
    cfg_path: Option<PathBuf>,
```

File: canbench-bin/src/main.rs [hunk 2]
```diff
@@ -53,6 +61,8 @@ fn main() {
             .unwrap_or(&DEFAULT_RESULTS_FILE.to_string()),
     );
 
+    let custom_drun_path = cfg.get("drun_path").map(PathBuf::from);
```
Comment (ielashi) [inline canbench-bin/src/main.rs:64]: I'm not a fan of adding `drun_path` to the configuration. The fact that we're using `drun` is an implementation detail, and we already have plans to swap out `drun` in favor of `PockerIC`, so adding it explicitly would expose it in the interface. Is this a necessary feature to add for bazel to work?

File: canbench-bin/src/main.rs (hunk 1 above)
Comment (jasonz-dfinity) [reply canbench-bin/src/main.rs]: Done.

File: canbench-bin/src/main.rs (hunk 2 above)
Comment (jasonz-dfinity) [reply canbench-bin/src/main.rs]: I see. It does impose some restrictions on using canbench as any `bazel build` step (AFAIK there is no network access during any `build`), but it seems OK just to run a benchmark using `bazel run`. Considering drun is being replaced it seems OK to revert the drun related changes for now.

Comment (jasonz-dfinity) [reply canbench-bin/src/main.rs]: I see. It does impose some restrictions on using canbench as any `bazel build` step (AFAIK there is no network access during any `build`), but it seems OK just to run a benchmark using `bazel run`. Considering drun is being replaced it seems OK to revert the drun related changes for now.

**Comment by ielashi** [comment]
LGTM. I've been looking into how to fix CI to work for forked repos - haven't succeeded yet unfortunately.

**Comment by jasonz-dfinity** [comment]
> LGTM. I've been looking into how to fix CI to work for forked repos - haven't succeeded yet unfortunately.

Would it help if I create a PR from a branch in this repo?

**Comment by jasonz-dfinity** [comment]
Closing this as I found a way to run canbench in ic monorepo without the custom file path

**Comment by ielashi** [comment]
I was actually in the middle of fixing the issue with forked repos (see #43). Sounds good. Would be happy to merge this PR if it ever becomes a need again.

**Pull request closed without merging**